src/services/BavService.ts

```diff
@@ -0,0 +1,182 @@
+			TableName: this.tableName,
+			Key: {
+				sessionId,
+			},
+		});
+
+		let session;
+		try {
+			session = await this.dynamo.send(getSessionCommand);
+		} catch (error) {
+			this.logger.error({ message: "getSessionById - failed executing get from dynamodb:" }, {
+				messageCode: MessageCodes.FAILED_FETCHING_SESSION,
+				error,
+			});
+			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Error retrieving Session");
+		}
+
+		if (session.Item) {
+			if (session.Item.expiryDate < absoluteTimeNow()) {
+				this.logger.error({ message: `Session with session id: ${sessionId} has expired` }, { messageCode: MessageCodes.EXPIRED_SESSION });
+				throw new AppError(HttpCodesEnum.UNAUTHORIZED, `Session with session id: ${sessionId} has expired`);
+			}
+			return session.Item as ISessionItem;
+		}
+	}
+
+	async sendToTXMA(event: TxmaEvent): Promise<void> {
+		try {
+			const messageBody = JSON.stringify(event);
+			const params = {
+				MessageBody: messageBody,
+				QueueUrl: process.env.TXMA_QUEUE_URL,
+			};
+
+			this.logger.info({ message: "Sending message to TxMA", eventName: event.event_name });
+
+			await sqsClient.send(new SendMessageCommand(params));
+			this.logger.info("Sent message to TxMA");
+		} catch (error) {
+			// highlight in the PR that I changed this
+			this.logger.error({
+				message: `Error when sending event ${event.event_name} to TXMA Queue`,
+				error,
+				messageCode: MessageCodes.FAILED_TO_WRITE_TXMA,
+			});
+		}
+	}
+
+	private mapBirthDate(birthDate: PersonIdentityDateOfBirth[]): PersonIdentityDateOfBirth[] {
+		return birthDate?.map((bd) => ({ value: bd.value }));
+	}
+
+	private mapNames(name: PersonIdentityName[]): PersonIdentityName[] {
+		return name?.map((index) => ({
+			nameParts: index?.nameParts?.map((namePart) => ({
+				type: namePart.type,
+				value: namePart.value,
+			})),
+		}));
+	}
+
+	private createPersonIdentityItem(
+		sharedClaims: SharedClaimsPersonIdentity,
+		sessionId: string,
+	): PersonIdentityItem {
+
+		const authSessionTtlInSecs: string | undefined = process.env.AUTH_SESSION_TTL_SECS;
+  	if (!authSessionTtlInSecs) {
+  		this.logger.error({
+  			message: "Missing AUTH_SESSION_TTL_SECS environment variable",
+  			messageCode: MessageCodes.MISSING_CONFIGURATION,
+  		});
+			throw new AppError(HttpCodesEnum.SERVER_ERROR, "Missing configuration" );
+  	}
+
+		return {
+			sessionId,
+			birthDate: this.mapBirthDate(sharedClaims.birthDate),
+			emailAddress: sharedClaims.emailAddress,
+			name: this.mapNames(sharedClaims.name),
+  		expiryDate: absoluteTimeNow() + +authSessionTtlInSecs,
+			createdDate: absoluteTimeNow(),
+		};
+	}
+
+	async savePersonIdentity(
+		sharedClaims: SharedClaimsPersonIdentity,
+		sessionId: string,
+	): Promise<void> {
+		const personIdentityItem = this.createPersonIdentityItem(
+			sharedClaims,
+			sessionId,
+		);
+
+		const putSessionCommand = new PutCommand({
+			TableName: process.env.PERSON_IDENTITY_TABLE_NAME,
```

Comment:
I do not see PERSON_IDENTITY_TABLE_NAME added as env var in template.yaml file. If we are using this function in session handler/processor then it may fail. This was one of the reasons why we had to have a way to validate if all the ENV var used for a service (handler) is defined/exists before we start the handler processing instead of failing at later state. We knew then that EnvironmentVaraibles.ts code is not optimised/right way and had to be re-thought and refactored, but we didnt get time to do that. But happy to have a call with the team and @ViveakV  to brainstorm this.

Reply:
Nice spot, added it in! Have put in time for us to chat about this later